protocol.go

```diff
@@ -59,7 +59,12 @@ func ReadResponse(r io.Reader) ([]byte, error) {
 
 	if msgSize < 0 {
 		return nil, fmt.Errorf("response msg size is negative: %v", msgSize)
+	} else if msgSize == 1213486160 { // Integer value of deserializing "HTTP"
+		return nil, fmt.Errorf("unexpected HTTP response, a nsqd TCP endpoint is required")
+	} else if msgSize > maxMsgSize {
```

Comment:
need to add a check here that `maxMsgSize > 0` too

Reply:
Done! The new code looks like this:

```go
if msgSize < 0 {
	return nil, fmt.Errorf("response msg size is negative: %v", msgSize)
}

if maxMsgSize > 0 && msgSize > maxMsgSize {
	if msgSize == httpResponseMsgSize {
		return nil, fmt.Errorf("unexpected HTTP response, a nsqd TCP endpoint is required")
	}
	return nil, fmt.Errorf("response msg size %v exceeds configured maximum (%v)", msgSize, maxMsgSize)
}
```

The HTTP response detection logic won't get in the way, i.e: if by chance a message exactly as big as `1213486160` bytes is read and the configured limit is above that value (or `0`), the function won't fail.